### PR TITLE
Issue/94 enable no redeclare

### PR DESCRIPTION
**Pull request by jitsedesmet**
This pull request aims to resolve most of #94. Except for the files `lib/functions/Helpers.ts` and `lib/functions/Core.ts`. These files cover the problem also talked about in #94 regarding the `OverloadMap`. Rewriting `OverloadMap` will be a pull request on itself.

### CODE REVIEW

File: lib/Types.ts [hunk 1]
```diff
@@ -9,8 +8,8 @@ export type Bindings = Map<string, RDF.Term>;
  * @return {Bindings} The immutable bindings from the hash.
  * @constructor
  */
-export function Bindings(hash: Record<string, RDF.Term>): Bindings {
-  return Map(hash);
+export function BindingsFromHash(hash: Record<string, RDF.Term>): Bindings {
```
Comment (rubensworks) [inline lib/Types.ts:11]: This (`Bindings`) is apparently literally taken over from Comunica.
Not sure how strict the relationship to Comunica is with respect to this, but can you double-check if everything still works after this change?

In any case, I see no need to rename this function.

File: lib/functions/NamedFunctions.ts [hunk 2]
```diff
@@ -27,7 +25,7 @@ type Term = E.TermExpression;
 // https://www.w3.org/TR/xpath-functions/#casting-from-primitive-to-primitive
 // ----------------------------------------------------------------------------
 
-const toString = {
+const toStringNamed = {
```
Comment (rubensworks) [inline lib/functions/NamedFunctions.ts:28]: Why did you rename these?

File: lib/functions/RegularFunctions.ts [hunk 3]
```diff
@@ -171,50 +170,50 @@ const greaterThanEqual = {
 // https://www.w3.org/TR/sparql11-query/#func-rdfTerms
 // ----------------------------------------------------------------------------
 
-const isIRI = {
+const rdfTermIsIRI = {
```
Comment (rubensworks) [inline lib/functions/RegularFunctions.ts:173]: Same as above, rename is not needed?

File: lib/functions/RegularFunctions.ts [hunk 4]
```diff
@@ -793,4 +792,4 @@ export interface IDefinition {
   overloads: OverloadMap;
 }
 
-export const definitions = Map<C.RegularOperator, IDefinition>(_definitions);
+export const definitions = _definitions;
```
Comment (rubensworks) [inline lib/functions/RegularFunctions.ts:795]: Just export the definitions above?

**Review by rubensworks — CHANGES_REQUESTED** [review]

File: lib/Types.ts (hunk 1 above)
Comment (jitsedesmet) [reply lib/Types.ts:11]: Well, the linter doesn't like a function Bindings and a type Bindings to have the same name. it does seem like comunica has some problems with this PR tho. I should investigate that closer.

File: lib/functions/NamedFunctions.ts (hunk 2 above)
Comment (jitsedesmet) [reply lib/functions/NamedFunctions.ts:28]: I don't really know, where but according to the linter this function already exists in the global-scope.

File: lib/functions/RegularFunctions.ts (hunk 3 above)
Comment (jitsedesmet) [reply lib/functions/RegularFunctions.ts:173]: This one is indeed not needed, but, since there was also a `toString` function (now `rdfTermToString`) I renamed all the functions so they still have the same naming style.

File: lib/functions/RegularFunctions.ts (hunk 4 above)
Comment (jitsedesmet) [reply lib/functions/RegularFunctions.ts:795]: Yes, we could do that.

File: lib/Types.ts (hunk 1 above)
Comment (jitsedesmet) [reply lib/Types.ts:13]: Seems like `BindingsFromHash` Isn't used in comunica. `Bindings` however is (in the config object, within exists). So I just made that an immutable Map again.

Comment (rubensworks) [reply lib/Types.ts:13]: Ok, good!

Could you restore the implementation as well to what it was before?

And I would recommend keeping the name `Bindings` for consistency with Comunica. It's ok to add a linter exception for this.

Comment (jitsedesmet) [reply lib/Types.ts:13]: Okey!

File: lib/functions/RegularFunctions.ts (hunk 3 above)
Comment (wschella) [reply lib/functions/RegularFunctions.ts:173]: In this case I would not rename any, except `toString` and just document that it's an en exception. I would say use `termToString` (rdf is unneeded here), `toString_`, or even `str` which is what it's called in SPARQL.

**Review by wschella — CHANGES_REQUESTED** [review]

**Review by rubensworks — APPROVED** [review]
Great work @jitsedesmet!
